res/controllers/Numark Mixtrack Pro FX.midi.xml

```diff
@@ -0,0 +1,1736 @@
+        </options>
+      </control>
+      <control>
+        <group>[Channel2]</group>
+        <key>MixtrackProFX.deck[1].syncButton.input</key>
+        <status>0x91</status>
+        <midino>0x02</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <!-- Sync (release) -->
+      <control>
+        <group>[Channel1]</group>
+        <key>MixtrackProFX.deck[0].syncButton.input</key>
+        <status>0x80</status>
+        <midino>0x02</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel2]</group>
+        <key>MixtrackProFX.deck[1].syncButton.input</key>
+        <status>0x81</status>
+        <midino>0x02</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <!-- Sync (shift) -->
+      <control>
+        <group>[Channel1]</group>
+        <key>MixtrackProFX.deck[0].syncButton.input</key>
+        <status>0x90</status>
+        <midino>0x03</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel2]</group>
+        <key>MixtrackProFX.deck[1].syncButton.input</key>
+        <status>0x91</status>
+        <midino>0x03</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <!-- CUE buttons -->
+      <control>
+        <group>[Channel1]</group>
+        <key>MixtrackProFX.deck[0].cueButton.input</key>
+        <status>0x90</status>
+        <midino>0x01</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel2]</group>
+        <key>MixtrackProFX.deck[1].cueButton.input</key>
+        <status>0x91</status>
+        <midino>0x01</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel1]</group>
+        <key>MixtrackProFX.deck[0].cueButton.input</key>
+        <status>0x80</status>
+        <midino>0x01</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel2]</group>
+        <key>MixtrackProFX.deck[1].cueButton.input</key>
+        <status>0x81</status>
+        <midino>0x01</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel1]</group>
+        <key>MixtrackProFX.deck[0].cueButtonShift.input</key>
+        <status>0x90</status>
+        <midino>0x05</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>
+      <control>
+        <group>[Channel2]</group>
+        <key>MixtrackProFX.deck[1].cueButtonShift.input</key>
+        <status>0x91</status>
+        <midino>0x05</midino>
+        <options>
+          <script-binding/>
+        </options>
+      </control>

```

Comment:
I didn't think they are needed, as the button works as intended. Should I add note-offs?

Reply:
Generally you should add note-offs for all applicable controls. Otherwise you might be confused when you change the mapping in the future and the button "gets stuck" because you didn't map the note-off signal.

Reply:
@Swiftb0y please take a look at the current code. I've added note-offs for all buttons, however in some cases I needed to add a condition that checks the value (e.g. in case of scratch toggle button, which should not react to note-off in any way). Is there a better way to handle this?

Reply:
Not really. It makes the mapping easier to understand in the long term because you can clearly see that the handler filters the event instead of the XML. The improved consistency is worth having a bit of overhead in the JS.